### PR TITLE
feat: add slider for adjusting traffic density

### PR DESCRIPTION
**Pull request by mozhoku**
## Description

This PR adds a slider to the UI for adjusting the traffic density.

Adding more NPC vehicles to the simulation heavily effects the performance. I've set the maximum value for the slider to 100. It can be modified from the editor if needed.

### Related links
 
- https://github.com/autowarefoundation/AWSIM/issues/64

## Tests performed

Tested in the editor:
- https://youtu.be/zH2sNbivYlo

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by mozhoku** [comment]
Demonstration:

- https://youtu.be/zH2sNbivYlo

File: Assets/AWSIM/Scripts/RandomTraffic/TrafficManager.cs [hunk 1]
```diff
@@ -22,28 +22,26 @@ public class TrafficManager : MonoBehaviour
         private LayerMask groundLayerMask;
 
         [SerializeField, Tooltip("A maximum number of vehicles that can simultaneously live in the scene. Lowering this value results in less dense traffic but improves the simulator's performance.")]
-        public int maxVehicleCount = 40;
+        public int maxVehicleCount = 100;
+        public int targetVehicleCount = 10;
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/RandomTraffic/TrafficManager.cs:26]:
```suggestion
        public int MaxVehicleCount = 100;
        public int TargetVehicleCount = 10;
```

File: Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Globalization;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace AWSIM.Scripts.UI
+{
+    public class UITrafficVehicleDensity : MonoBehaviour
+    {
+        [SerializeField] private Text textField;
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs]:
```suggestion
        [SerializeField] private Text _textField;
```

**Review by xmfcx — CHANGES_REQUESTED** [review]
Some naming guideline suggestions.

File: Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs (hunk 2 above)
Comment (mozhoku) [reply Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs]: Rider does not suggests "_" prefix to the "[SerializeField] private xxx yyyy" variables

Comment (xmfcx) [reply Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs]: https://docs.unity3d.com/Manual/script-Serialization.html

Unity treats them same as private members. So apply the underscore please.

Rider probably learns the conventions from the code itself.

File: Assets/AWSIM/Scripts/RandomTraffic/TrafficManager.cs (hunk 1 above)
Comment (mozhoku) [reply Assets/AWSIM/Scripts/RandomTraffic/TrafficManager.cs:26]: https://github.com/autowarefoundation/AWSIM/pull/67/commits/b82e26297e8469efed2120089ac58232b32e2f67

File: Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs (hunk 2 above)
Comment (mozhoku) [reply Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs]: https://github.com/autowarefoundation/AWSIM/pull/67/commits/65bfe1d54f9f629c3c9b3f3bcac2a2e6be23cf61

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/AWSIM/assets/10751153/c89f2a14-b0a7-4553-93d6-2e566abef391

```
NullReferenceException: Object reference not set to an instance of an object
AWSIM.Scripts.UI.UITrafficVehicleDensity.TrafficManagerSetTargetVehicleCount () (at Assets/AWSIM/Scripts/UI/UITrafficVehicleDensity.cs:27)
UnityEngine.Events.InvokableCall.Invoke () (at /home/bokken/build/output/unity/unity/Runtime/Export/UnityEvent/UnityEvent.cs:178)
UnityEngine.Events.UnityEvent`1[T0].Invoke (T0 arg0) (at /home/bokken/build/output/unity/unity/artifacts/generated/UnityEvent/UnityEvent_1.cs:62)
UnityEngine.UI.Slider.Set (System.Single input, System.Boolean sendCallback) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/UI/Core/Slider.cs:540)
UnityEngine.UI.Slider.set_value (System.Single value) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/UI/Core/Slider.cs:267)
UnityEngine.UI.Slider.set_normalizedValue (System.Single value) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/UI/Core/Slider.cs:314)
UnityEngine.UI.Slider.UpdateDrag (UnityEngine.EventSystems.PointerEventData eventData, UnityEngine.Camera cam) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/UI/Core/Slider.cs:623)
UnityEngine.UI.Slider.OnDrag (UnityEngine.EventSystems.PointerEventData eventData) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/UI/Core/Slider.cs:657)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IDragHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/ExecuteEvents.cs:78)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/ExecuteEvents.cs:272)
UnityEngine.EventSystems.EventSystem:Update() (at ./Library/PackageCache/com.unity.ugui@1.0.0/Runtime/EventSystem/EventSystem.cs:530)
```

**Comment by mozhoku** [comment]
@xmfcx After renaming the SerializedField, reference is lost. Should be okay now.

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/AWSIM/assets/10751153/a5db774d-5c21-4947-a72d-fdc23cc0ff0c

It works :+1:

**Review by xmfcx — APPROVED** [review]

**Pull request merged**